keras_cv/layers/preprocessing/random_flip_test.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+
+import numpy as np
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers.preprocessing.random_flip import RandomFlip
+
+
+class RandomFlipTest(tf.test.TestCase, parameterized.TestCase):
+    @parameterized.named_parameters(
+        ("random_flip_horizontal", "horizontal"),
+        ("random_flip_vertical", "vertical"),
+        ("random_flip_both", "horizontal_and_vertical"),
+    )
+    def test_random_flip(self, mode):
+        np.random.seed(1337)
+        mock_random = [0.6, 0.6]
+        if mode == "horizontal_and_vertical":
+            mock_random *= 2
+        inp = np.random.random((2, 5, 8, 3))
+        expected_output = inp
+        if mode == "horizontal" or mode == "horizontal_and_vertical":
```

Comment:
Avoid logic in unit tests: for example, if you find yourself having multiple branches to craft expected output you should create an extra test case.  In this case, I recommend splitting into three cases to avoid conditional logic in the test case.

You can read more about this here:
go/unit-testing-practices#logic
(sorry community members for the internal link)

Reply:
Looks like a complete refactoring of legacy code from Keras  imagepreprocessinglayer.py  is required 😅 . I tried to avoid making changes to older code while migrating. but looks like the coding style was not followed for the Keras preprocessing layers too.